pallets/testing-utilities/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+
+pub type ScheduledTransferOf<T> =
+	ScheduledTransfer<<T as frame_system::Config>::AccountId, <T as Config>::Balance>;
+
+pub type BlockNumberFor<T> =
+	<<T as Config>::BlockNumberProvider as BlockNumberProvider>::BlockNumber;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	use codec::HasCompact;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_balances::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type Currency: Inspect<Self::AccountId, Balance = <Self as Config>::Balance>
+			+ Mutate<Self::AccountId>;
+
+		type Balance: Balance
+			+ HasCompact<Type: DecodeWithMemTracking>;
+
+		type BlockNumberProvider: BlockNumberProvider;
+
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		Scheduled,
+		Executed { scheduled_in: BlockNumberFor<T> },
+	}
+
+	#[pallet::storage]
+	pub type ScheduledTransfers<T: Config> =
+		StorageMap<_, Twox64Concat, BlockNumberFor<T>, ScheduledTransferOf<T>, OptionQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Source balance is too low to transfer the required amount.
+		SourceBalanceTooLow,
+		/// Destination balance is too low even after transfer.
+		DestinationBalanceTooLow,
+		/// Unable to schedule due to agenda being full,
+		CouldNotSchedule,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Schedules on the next block a `burn` of the funds on the `from` wallet
+		/// with immediate `mint` on the `to` wallet.
+		///
+		/// # Arguments
+		/// * `origin` - The origin of the transaction, source account for transfer.
+		/// * `to` - Destination account for the balance transfer.
+		/// * `amount` - The amount to be burned and reminted.
+		///
+		/// # Errors
+		/// * `Error::<T>::SourceBalanceTooLow` when `from` balance is too low
+		///   to transfer the required amount.
+		/// * `Error::<T>::DestinationBalanceTooLow` when `to` balance will be
+		///   below minimum after transfer.
+		/// * `Error::<T>::CouldNotSchedule` when another task was already
+		///   scheduled in this block.
+		#[pallet::weight(<T as Config>::WeightInfo::transfer_through_delayed_remint())]
+		#[pallet::call_index(0)]
+		pub fn transfer_through_delayed_remint(
+			origin: OriginFor<T>,
+			to: T::AccountId,
+			#[pallet::compact] amount: BalanceOf<T>,
+		) -> DispatchResult {
+			let from = ensure_signed(origin.clone())?;
+
+			ensure!(
+				T::Currency::balance(&from).saturating_sub(amount)
+					>= T::Currency::minimum_balance(),
+				<Error<T>>::SourceBalanceTooLow,
+			);
+
+			ensure!(
+				T::Currency::balance(&to).saturating_add(amount) >= T::Currency::minimum_balance(),
+				<Error<T>>::DestinationBalanceTooLow,
+			);
+
+			let current_block = T::BlockNumberProvider::current_block_number();
+
+			ensure!(
+				!<ScheduledTransfers<T>>::contains_key(&current_block),
```

Comment:
Out of curiosity: why only one operation per block? I understand this is not crucial, as it is just for testnet... but analyzing coldly, an attacker could simply attempt to fill the available single-block space for transfers in each and every block.